homeassistant/components/elmax/__init__.py

```diff
@@ -21,17 +38,75 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+async def _load_elmax_panel_client(
+    entry: ConfigEntry,
+) -> tuple[GenericElmax, PanelEntry]:
+    # Connection mode was not present in initial version, default to cloud if not set
+    mode = entry.data.get(CONF_ELMAX_MODE, CONF_ELMAX_MODE_CLOUD)
+    if mode == CONF_ELMAX_MODE_CLOUD:
+        client = Elmax(
+            username=entry.data[CONF_ELMAX_USERNAME],
+            password=entry.data[CONF_ELMAX_PASSWORD],
+        )
+        client.set_current_panel(
+            entry.data[CONF_ELMAX_PANEL_ID], entry.data[CONF_ELMAX_PANEL_PIN]
+        )
+        # Make sure the panel is online and assigned to the current user
+        panel = await _check_cloud_panel_status(client, entry.data[CONF_ELMAX_PANEL_ID])
+    elif mode == CONF_ELMAX_MODE_DIRECT:
+        client_api_url = get_direct_api_url(
+            host=entry.data[CONF_ELMAX_MODE_DIRECT_HOST],
+            port=entry.data[CONF_ELMAX_MODE_DIRECT_PORT],
+            use_ssl=entry.data[CONF_ELMAX_MODE_DIRECT_SSL],
+        )
+        custom_ssl_context = None
+        custom_ssl_cert = entry.data.get(CONF_ELMAX_MODE_DIRECT_SSL_CERT)
+        if custom_ssl_cert:
+            custom_ssl_context = build_direct_ssl_context(cadata=custom_ssl_cert)
+
+        client = ElmaxLocal(
+            panel_api_url=client_api_url,
+            panel_code=entry.data[CONF_ELMAX_PANEL_PIN],
+            ssl_context=custom_ssl_context,
+        )
+        panel = DirectPanel(panel_uri=client_api_url)
+    else:
+        raise ConfigEntryAuthFailed(
+            f"Invalid configuration detected. Unsupported mode={mode}"
+        )
```

Comment:
I'm not convinced this is needed

Reply:
Done, handled in 15138ff56967cf0b09f836ef427cb1f0cb32754e